### ISSUE TITLE
Server configure command changes are not saved if the value is empty

### ISSUE THREAD

**Issue by thebigmoosey** [OctopusDeploy/Issues#3451]
Following the instructions for Calamari, resetting the `customBundledPackageDirectory` no longer works. Eg:

```
Octopus.Server.exe configure --instance <instance> --customBundledPackageDirectory= --nologo --console
```

On investigation, we have an early-exit condition in the Setter for `AbstractKeyValueStore` when checking for nullOrWhiteSpace and the `autoSaveOnSet` property is not checked here when it should be: Eg. 
https://github.com/OctopusDeploy/OctopusDeploy/blob/master/source/Octopus.Shared/Configuration/AbstractKeyValueStore.cs#L59

**Comment by matt-richardson** [comment]
I wonder if thats why PrimaryOctopusServer.cs#L37 does it by directly injecting into the config file?

**Comment by thebigmoosey** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [fix abstractKeyValue issue in release branch (this will then automatically get merged back into master)](https://trello.com/c/l5cMjxb5/5-fix-abstractkeyvalue-issue-in-release-branch-this-will-then-automatically-get-merged-back-into-master)

**Comment by thebigmoosey** [comment]
This was fixed and rolled out as part of 3.13
